src/QsCompiler/SyntaxProcessor/TypeInference/InferenceContext.fs

```diff
@@ -288,6 +288,8 @@ open Inference
 type InferenceContext(symbolTracker: SymbolTracker) =
     let variables = Dictionary()
 
+    let mutable rootNodePos = Null
+    let mutable relativePos = Null
```

Comment:
It seems as if with the addition of these two members, `statementPostion` just became a proxy for `rootNodePosition + relativePosition`. If that is the case, should `statementPosition` be removed/deprecated?

Reply:
statementPosition is a proxy for rootNodePosition + relativePosition if a location has been specified, but could be set and used on its own without ever specifying the location broken down into absolute and relative position (in which case the handle to get the relative location throws). See also my comment above.